include/fi_util.h

```diff
@@ -295,6 +295,38 @@ struct util_event {
 int fi_eq_create(struct fid_fabric *fabric, struct fi_eq_attr *attr,
 		 struct fid_eq **eq_fid, void *context);
 
+/*
+ * Buffer Pool
+ */
+
+struct util_buf_pool {
+	size_t size;
+	size_t max_num;
+	size_t chunk_num;
+	size_t alignment;
+	size_t num_allocated;
+	size_t num_used;
+
+	struct slist buf_list;
+	struct slist region_list;
+};
+
+struct util_buf_region {
+	struct slist_entry entry;
+	char *mem_region;
+};
+
+union util_buf {
+	struct slist_entry entry;
+	uint8_t data[0];
+};
+
+struct util_buf_pool *util_buf_pool_create(size_t size, size_t alignment,
+					   size_t max_num, size_t chunk_num);
```

Comment:
I prefer cnt versus num.  chunk_num sounds like it refers to a specific chunk, rather than a count or size